mplc/multi_partner_learning.py

```diff
@@ -603,12 +603,64 @@ def fit_minibatch(self):
         logger.debug("End of S-Model collaborative round.")
 
 
-# Supported multi-partner learning approaches
+class EnsemblePredictions(MultiPartnerLearning):
+    """
+    Ensemble (average) prediction of several input models
+    This approach can only be used with the EnsemblePredictionsModel
+    """
+
+    def __init__(self, scenario, **kwargs):
+        # First, if only one partner, fall back to dedicated single partner function
+        super(EnsemblePredictions, self).__init__(scenario, **kwargs)
+
+        if self.partners_count == 1:
+            raise ValueError('Only one partner is provided. Please use the dedicated SinglePartnerLearning class')
+
+        partner_model_list = [self.dataset.generate_new_model() for _ in range(self.partners_count)]
+        self.model = EnsemblePredictionsModel(partner_model_list)
+        # Issue here is that we have 2 models, the Ensemble model and
+        # the model for every partner. self.model could either be the EnsembleModel
+        # or the partner model but in both cases it's not super straightforward.
+
```

Comment:
@arthurPignet help :)

Reply:
I think that the mpl.model should be the Ensemble model, whihc hold as attribute a list of submodel (one per partner)

Therefore in the .fit method, you could do somethink like :

```python
for p in self. partner_list:
      self.model.partners_models[p.id].fit(p.x_train, p.y_train)

```